### PR TITLE
chore: set up storybook (#5191)

### LINKED ISSUE #5191: Adopt Storybook for the Component Migration
As part of the [Website Redesign Migration](https://github.com/nodejs/nodejs.org/discussions/5131) we want to kick off things by migrating our Components from `nodejs.dev` to `nodejs.org`.

Of course, as the Layouts and Pages are only going to be migrated later, we need another way to preview the Components so we can design-review them and unit-test them.

Hence we should adopt Storybook on our `major/website-redesign` branch.

This issue involves implementing [Storybook](https://storybook.js.org/) on our Next.js repository. (There's a guide here https://storybook.js.org/blog/integrate-nextjs-and-storybook-automatically/)

Anyone, feel free to tackle this issue.

### PR DESCRIPTION
**Pull request by araujogui**
Set up Storybook and create two testing stories.

Issue: #5191

### CODE REVIEW

**Review by ovflowd — APPROVED** [review]
LGTM! Really good first contribution. Thank you for working on this! 🙇

File: package.json [hunk 1]
```diff
@@ -79,6 +88,7 @@
     "remark-preset-lint-node": "^3.4.0",
     "remark-preset-prettier": "^2.0.1",
     "sharp": "^0.31.3",
+    "storybook": "^7.0.0-rc.8",
```
Comment (mikeesto) [inline package.json:91]: Just curious as to why we are using the release candidate? The latest version atm seems to be 6.5.16

Comment (ovflowd) [reply package.json:91]: Looks like Storybook's own guides (and CLI tooling for setting up Storybook) chose this version.

**Review by aymen94 — APPROVED** [review]

**Comment by aymen94** [comment]
Awesome first contribution! 😀

File: package.json (hunk 1 above)
Comment (manishprivet) [reply package.json:91]: We can probably use it here as well if this RC is going to be released soon, will save troubles of migration

**Review by manishprivet — APPROVED** [review]

**Review by shanpriyan — APPROVED** [review]
LGTM! Good stuff. Thanks for the contribution ⭐️

**Review by mikeesto — APPROVED** [review]

**Pull request merged**